chain/chaintest/block.go

```diff
@@ -42,3 +55,81 @@ func GenerateEmptyExecutedBlocks(
 	}
 	return executedBlocks
 }
+
+// GenerateTestExecutedBlocks creates [numBlocks] executed block for testing purposes, each with [numTx] transactions.
+// these blocks are not semantically valid for execution, and are meant execlusively for serialization testing purposes.
+func GenerateTestExecutedBlocks(
+	require *require.Assertions,
+	parentID ids.ID,
+	chainID ids.ID,
+	parentHeight uint64,
+	parentTimestamp int64,
+	timestampOffset int64,
+	numBlocks int,
+	numTx int,
+) []*chain.ExecutedBlock {
+	bh := balance.NewPrefixBalanceHandler([]byte{0})
+	rules := genesis.NewDefaultRules()
+	feeManager := internal_fees.NewManager(nil)
+	executedBlocks := make([]*chain.ExecutedBlock, numBlocks)
+	parser := NewTestParser()
+	ts := tstate.New(0)
+	db, err := merkledb.New(
+		context.Background(),
+		memdb.New(),
+		merkledb.Config{BranchFactor: merkledb.BranchFactor16},
+	)
+	require.NoError(err)
+
+	tstateview := ts.NewView(state.CompletePermissions, db, 0)
+	require.NoError(tstateview.Insert(context.Background(), balance.NewPrefixBalanceHandler([]byte{0}).BalanceKey(NewDummyTestAuth().Sponsor()), binary.BigEndian.AppendUint64(nil, math.MaxUint64)))
+
```

Comment:
Can we create one instance of the balance handler and call the necessary functions on that rather than create an instance and immediately call a function on it and re-use the constructor in different places?

This is error prone if we change one instance, but not the others and less readable